eureka/src/main/java/com/linecorp/armeria/client/eureka/EurekaEndpointGroupBuilder.java

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.eureka;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.client.AbstractClientOptionsBuilder;
+import com.linecorp.armeria.client.ClientFactory;
+import com.linecorp.armeria.client.ClientOption;
+import com.linecorp.armeria.client.ClientOptionValue;
+import com.linecorp.armeria.client.ClientOptions;
+import com.linecorp.armeria.client.DecoratingHttpClientFunction;
+import com.linecorp.armeria.client.DecoratingRpcClientFunction;
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.client.RpcClient;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.client.WebClientBuilder;
+import com.linecorp.armeria.client.endpoint.EndpointGroup;
+import com.linecorp.armeria.common.RequestId;
+import com.linecorp.armeria.common.auth.BasicToken;
+import com.linecorp.armeria.common.auth.OAuth1aToken;
+import com.linecorp.armeria.common.auth.OAuth2Token;
+
+/**
+ * Builds a {@link EurekaEndpointGroup}.
+ */
+public final class EurekaEndpointGroupBuilder extends AbstractClientOptionsBuilder {
+
+    private static final long DEFAULT_REGISTRY_FETCH_INTERVAL_SECONDS = 30;
+
+    private final URI eurekaUri;
+
+    @Nullable
+    private String appName;
+
+    @Nullable
+    private String instanceId;
+
+    @Nullable
+    private String vipAddress;
+
+    @Nullable
+    private String secureVipAddress;
+
+    private long registryFetchIntervalSeconds = DEFAULT_REGISTRY_FETCH_INTERVAL_SECONDS;
+
+    @Nullable
+    private List<String> regions;
+    @Nullable
+    private Consumer<WebClientBuilder> customizer;
+
+    EurekaEndpointGroupBuilder(URI eurekaUri) {
```

Comment:
Question: Don't we need to take multiple eureka URIs? From the HA spec of eureka, a client should attempt to connect other eureka servers if a eureka is sitting idle.
https://blog.asarkar.org/technical/netflix-eureka/#high-availability-ha

Reply:
That's a good idea. How about adding a constructor that takes an `EndpointGroup`?

Reply:
SGTM :-)